File: src/test/java/br/com/blz/testjava/model/InventoryTest.java
package br.com.blz.testjava.model;

import static br.com.blz.testjava.utils.ObjectCreator.getInventory;
import static org.assertj.core.api.Assertions.assertThat;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.test.context.junit4.SpringRunner;

@RunWith(SpringRunner.class)
public class InventoryTest {
	@Test
	public void testEqualsAndHashCodeSymmetric() {
		Inventory inventory1 = getInventory();
		Inventory inventory2 = getInventory();
		
		assertThat(inventory1.equals(inventory2)).isTrue();
		assertThat(inventory1.hashCode()).isEqualTo(inventory2.hashCode());
		assertThat(inventory1.toString()).isEqualTo(inventory2.toString());
	}
}

File: src/test/java/br/com/blz/testjava/model/InventoryTest.kt
package br.com.blz.testjava.model

import br.com.blz.testjava.enum.ProductType
import org.assertj.core.util.Lists
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class InventoryTest {
  private lateinit var warehouse:  Warehouse
  private lateinit var warehouse2: Warehouse
  private lateinit var inventory: Inventory

  @BeforeEach
  fun init() {
    warehouse2 = Warehouse("Rio de Janeiro", 5, ProductType.ECOMMERCE)
    warehouse = Warehouse("São Paulo", 2, ProductType.ECOMMERCE)
    inventory = Inventory(Lists.list(warehouse, warehouse2))
  }

  @Test
  fun inventoryTest() {
    val quantity = inventory.getQuantity()
    Assertions.assertEquals(quantity, 7)
  }
}
